packages/lesswrong/components/posts/PostsEditForm.tsx

```diff
@@ -18,57 +18,68 @@ import type { Editor } from '@ckeditor/ckeditor5-core';
 import { useNavigate } from '../../lib/reactRouterWrapper';
 import { preferredHeadingCase } from '../../themes/forumTheme';
 import DeferRender from '../common/DeferRender';
+import { useSingleWithPreload } from '@/lib/crud/useSingleWithPreload';
 
 const editor: Editor | null = null
 export const EditorContext = React.createContext<[Editor | null, (e: Editor) => void]>([editor, _ => {}]);
 
+function getDraftLabel(post: PostsPage | null) {
+  if (!post) return "Save Draft";
+  if (!post.draft) return "Move to Drafts";
+  return "Save Draft";
+}
+
 const PostsEditForm = ({ documentId, version, classes }: {
   documentId: string,
   version?: string | null,
   classes: ClassesType,
 }) => {
+  const { WrappedSmartForm, PostSubmit, SubmitToFrontpageCheckbox, HeadTags, ForeignCrosspostEditForm, DialogueSubmit, RateLimitWarning, DynamicTableOfContents } = Components
+
   const { query, params } = useLocation();
   const navigate = useNavigate();
   const { flash } = useMessages();
-  const { document, loading } = useSingle({
-    documentId,
-    collectionName: "Posts",
-    fragmentName: 'PostsPage',
-  });
   const { openDialog } = useDialog();
   const currentUser = useCurrentUser();
-  const isDraft = document && document.draft;
 
-  const wasEverDraft = useRef(isDraft);
-  useEffect(() => {
-    if (wasEverDraft.current === undefined && isDraft !== undefined) {
-      wasEverDraft.current = isDraft;
-    }
-  }, [isDraft]);
-
-  const { WrappedSmartForm, PostSubmit, SubmitToFrontpageCheckbox, HeadTags, ForeignCrosspostEditForm, DialogueSubmit, RateLimitWarning, DynamicTableOfContents } = Components
+  const [editorState, setEditorState] = useState<Editor | null>(editor);
 
-  const [editorState, setEditorState] = useState<Editor | null>(editor)
-  const saveDraftLabel: string = ((post) => {
-    if (!post) return "Save Draft"
-    if (!post.draft) return "Move to Drafts"
-    return "Save Draft"
-  })(document)
-  
-  const { mutate: updatePost } = useUpdate({
+  const { bestResult: document, fetchedResult: { loading } } = useSingleWithPreload({
```

Comment:
This and the new `editFormFetchPolicy` prop on the wrapped smart form are the only meaningful changes to this file; the rest was me moving things around because the previous state of affairs was making me very sad.

Reply:
I think is may be finally time to get rid of the New Post page, and have the Create Post button make an empty draft and send you to the Edit page. This isn't the first time that "editing a new post with no corresponding post in the DB" has been a case that drives complexity.